cvat/apps/iam/rules/limits.rego

```diff
@@ -0,0 +1,129 @@
+package limits
+
+import future.keywords.if
+import future.keywords.in
+import future.keywords.contains
+
+import data.utils
+
+
+CAP_USER_SANDBOX_TASKS = "USER_SANDBOX_TASKS"
+CAP_USER_SANDBOX_PROJECTS = "USER_SANDBOX_PROJECTS"
+CAP_TASKS_IN_USER_SANDBOX_PROJECT = "TASKS_IN_USER_SANDBOX_PROJECT"
+CAP_USER_OWNED_ORGS = "USER_OWNED_ORGS"
+CAP_USER_SANDBOX_CLOUD_STORAGES = "USER_SANDBOX_CLOUD_STORAGES"
+CAP_ORG_TASKS = "ORG_TASKS"
+CAP_ORG_PROJECTS = "ORG_PROJECTS"
+CAP_TASKS_IN_ORG_PROJECT = "TASKS_IN_ORG_PROJECT"
+CAP_ORG_CLOUD_STORAGES = "ORG_CLOUD_STORAGES"
+CAP_ORG_COMMON_WEBHOOKS = "ORG_COMMON_WEBHOOKS"
+CAP_PROJECT_WEBHOOKS = "PROJECT_WEBHOOKS"
+
+
+check_limit_exceeded(current, max) {
+    null != max
+    current >= max
+}
+
+
+
+problems contains "user tasks limit reached" if {
+    check_limit_exceeded(
+        input.resource.limits[CAP_USER_SANDBOX_TASKS].used,
+        input.resource.limits[CAP_USER_SANDBOX_TASKS].max
+    )
+}
+
+problems contains "user projects limit reached" if {
+    check_limit_exceeded(
+        input.resource.limits[CAP_USER_SANDBOX_PROJECTS].used,
+        input.resource.limits[CAP_USER_SANDBOX_PROJECTS].max
+    )
+}
+
+problems contains "user project tasks limit reached" if {
+    check_limit_exceeded(
+        input.resource.limits[CAP_TASKS_IN_USER_SANDBOX_PROJECT].used,
+        input.resource.limits[CAP_TASKS_IN_USER_SANDBOX_PROJECT].max
+    )
+}
+
+problems contains "org tasks limit reached" if {
```

Comment:
It is possible to use sprintf function to add information about current limits.

Reply:
sprintf("org projects limit reached. Current number of projects %d, maxim number of projects should be less than %d", [input.resource.limits[CAP_ORG_PROJECTS].used, input.resource.limits[CAP_ORG_PROJECTS].max])